### PR TITLE
feat: UM-34, UM-36 components button

### PR DESCRIPTION
**Pull request by ingrqm**
[UM-34 components button](https://github.com/users/Dhitpl/projects/1?pane=issue&itemId=77462288)
[UM-36 icon investigation](https://github.com/users/Dhitpl/projects/1?pane=issue&itemId=80700613)

## What was the task?
Create button component
Handle prefixIcon, suffixIcon, isLoading and disabled
Improve performance on icon rendering

## What has been changed?
- add breakpoints config to styles,
- mapped breakpoints to tailwind config
- button adjustments
- icon improvements
- use breakpoint hook
- util to convert case style
- utility type to convert case style

## How does it look like?

|                     Before                     |                     After                     |
| :--------------------------------------------: | :-------------------------------------------: |
| image | image |

## Checklist
- [x] The code has been cleaned up (unused imports, `console.log`s, etc.).
- [x] The PR has its assignee(s) selected.
- [x] The PR has its reviewer(s) selected.
- [x] The PR’s title is compliant with our `commit-lint` configuration.

### CODE REVIEW

File: src/hooks/use-breakpoint.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+/**
+ * Custom hook to determine if the current window width matches a specified breakpoint.
+ *
+ * @param {UseIsBreakpointProps} props - The properties object.
+ * @param {Breakpoint} props.breakpoint - The breakpoint to match against.
+ * @returns {boolean} - A boolean indicating if the current window width matches the specified breakpoint.
+ */
+export const useIsBreakpoint = ({ breakpoint }: UseIsBreakpointProps) => {
```
Comment (Juleww) [inline src/hooks/use-breakpoint.ts:18]: In a server-side rendered (SSR) environment, 'window' is not available during initial render.
Attempting to access 'window' directly can cause runtime errors in SSR contexts.

Suggestion: by adding
`if (typeof window === 'undefined') {
 return false // or some initial value like false
}`
we prevent this code from running on the server, ensuring it only runs in the browser.

**Review by BoNyPL — APPROVED** [review]
Great job!

File: src/hooks/use-breakpoint.ts (hunk 1 above)
Comment (ingrqm) [reply src/hooks/use-breakpoint.ts:18]: This file includes: `'use client'` on the top

There is no possibility to use it on server components so there is no need to cast window undefined

Comment (ingrqm) [reply src/hooks/use-breakpoint.ts:18]: If it cause problem during server render we will have build error

**Pull request merged**